### ISSUE TITLE
Job must be specified error with JSON job spec

### ISSUE THREAD

**Issue by cleung2010** [hashicorp/nomad#648]
I am encountering this error:

```bash 
$ curl -X POST -d @helloworld.json  http://127.0.0.1:4646/v1/jobs
Job must be specified
```

Using Kelsey's example helloworld job converted to JSON
```json
{
  "Region": "dc1",
  "ID": "helloworld",
  "Name": "helloworld",
  "Datacenters": [
    "dc1"
  ],
  "Update": {
    "Stagger": 5,
    "MaxParallel": 1
  },
  "TaskGroups": {
    "Name": "helloworld",
    "Count": 2,
    "Tasks": [
      {
        "Name": "helloworld",
        "Driver": "docker",
        "Config": {
          "image": "b.gcr.io/kuar/helloworld:1.0.0"
        },
        "Resources": {
          "CPU": 100,
          "MemoryMB": 200
        }
      },
      {
        "Name": "redis",
        "Driver": "docker",
        "Config": {
          "image": "redis:2.8.22"
        },
        "Resources": {
          "CPU": 100,
          "MemoryMB": 200
        }
      }
    ]
  }
}

```

https://github.com/hashicorp/nomad/blob/7ab84c2862d8f8de75e9ac64ee71b8a0cd05c798/command/agent/job_endpoint.go#L164

Edit: Fix "Update" hash.

**Comment by cleung2010** [comment]
Found a related post [GGroups](https://groups.google.com/forum/#!msg/nomad-tool/bVUYHSGqJL8/q72uCb7XDAAJ). 

According to that post, one of the missing things is "Job" in the outermost section. However, the API docs states that "The general structure is given by the job specification, and matches the return response of GET" , but the GET response does not contain that.

**Comment by diptanu** [comment]
@cleung2010 What do you mean by the outer section?

**Comment by cleung2010** [comment]
Also, I am assuming the syntax is PascalCase for the JSON object but "images" in "Config" is in underscore. What was the decision to use PascalCase on JSON but snake_case on HCL? Wouldn't it be more consistent (and further reduce mapping confusion) if the same convention was used for both cases?

**Comment by dadgar** [comment]
@cleung2010: The difference in case is just from trying to be idiomatic in each language.

**Comment by cleung2010** [comment]
I see. Was there a particular reason as to going with PascalCase on JSON than say camelCase? I don't think it matters much and it would be impractical to change the syntax now due backwards compatibility, but I was just curious about this.

**Comment by cleung2010** [comment]
Just tested the another parameter inside Config, `port_map`, which is also valid when used as-in (non-PascalCase). 

```
"Config": {
  "image": "b.gcr.io/kuar/helloworld:1.0.0",
  "port_map": {
    "http": 80
  }
}
```

I am not sure whether the other params on https://www.nomadproject.io/docs/drivers/docker.html didn't get their PascalCase/JSON equivalent, but these two are the ones I tested so far.

**Comment by jhartman86** [comment]
Trying to schedule jobs by sending JSON to the API, it seems like the JSON structure that gets accepted does not match up with what is in the documentation. Or more accurately, when running a job specified with JSON via the CLI it works, but posting the same JSON to the API fails. The response that gets returned seem to indicate that when run via the CLI, nomad is transforming it (probably same phase as if it were ingesting/transform HCL?) before using it.

For example, with a minimal job:
```
{
  "job": {
    "test": {
      "datacenters": ["local"],
      "group": {
        "gtest": {
          "task": [{
            "ttest": {
              "driver": "docker",
              "config": {
                "image": "smebberson/alpine-nginx"
              },
              "resources": {
                "cpu": 500,
                "memory": 64
              }
            }
          }]
        }
      }
    }
  }
}
```
^ Works with `$: nomad run job.nomad`. But when posting to the API (`curl -X POST -H 'Content-Type:application/json' --data @job.nomad $NOMAD_ADDR/v1/jobs`), it fails. API returns:
```
rpc error: 7 error(s) occurred:

* Missing job region
* Missing job ID
* Missing job name
* Missing job type
* Job priority must be between [1, 100]
* Missing job datacenters
* Missing job task groups
```

**Comment by dadgar** [comment]
Sorry for the delay on this guys! Just a lot of features being baked and issues being fixed. With the release of 0.3 we will update the documentation on submitting jobs via the JSON API

**Comment by jhartman86** [comment]
@dadgar :+1: no worries, project is shaping up great so far. Keep trucking.

**Comment by diptanu** [comment]
Fixed via #925

**Comment by diptanu** [comment]
We just added a section called Json Syntax in the documentation.
